eclass/pypi.eclass

```diff
@@ -63,6 +63,21 @@ _PYPI_ECLASS=1
 # @CODE
 : "${PYPI_PN:=${PN}}"
 
+# @FUNCTION: _pypi_normalize_name
+# @INTERNAL
+# @USAGE: <name>
+# @DESCRIPTION:
+# Internal normalization function, returns the result
+# via _PYPI_NORMALIZED_NAME variable.
+_pypi_normalize_name() {
+	local name=${1}
+	local shopt_save=$(shopt -p extglob)
+	shopt -s extglob
```

Comment:
You can save another fork:
```bash
local shopt_save=
if ! shopt -p extglob; then
    shopt_save='shopt -u extglob'
    shopt -s extglob
fi
do_stuff
$shopt_save
```

Reply:
Nice idea. I'll try later, when I have free CPU.

Reply:
Thanks, this is great! 550% speedup ;-).